pkg/controller/operators/olm/operator.go

```diff
@@ -415,29 +445,33 @@ func (a *Operator) removeDanglingChildCSVs(csv *v1alpha1.ClusterServiceVersion)
 	operatorNamespace, ok := csv.Annotations[v1alpha2.OperatorGroupNamespaceAnnotationKey]
 	if !ok {
 		logger.Debug("missing operator namespace annotation on copied CSV")
-		// TODO: Should we clean up the CSV if we don't know where its parent is?
-		return fmt.Errorf("missing operator namespace annotation on copied CSV")
+		return a.deleteChild(csv)
 	}
 
-	delete := false
 	parent, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(operatorNamespace).Get(csv.GetName())
-	if k8serrors.IsNotFound(err) || k8serrors.IsGone(err) {
+	if k8serrors.IsNotFound(err) || k8serrors.IsGone(err) || parent == nil {
 		logger.Debug("deleting copied CSV since parent is missing")
-		delete = true
-	} else if parent != nil && parent.Status.Phase == v1alpha1.CSVPhaseFailed && parent.Status.Reason == v1alpha1.CSVReasonInterOperatorGroupOwnerConflict {
+		return a.deleteChild(csv)
+	}
+	if parent.Status.Phase == v1alpha1.CSVPhaseFailed && parent.Status.Reason == v1alpha1.CSVReasonInterOperatorGroupOwnerConflict {
 		logger.Debug("deleting copied CSV since parent has intersecting operatorgroup conflict")
-		delete = true
-	} 
+		return a.deleteChild(csv)
+	}
 
-	if delete {
-		if err := a.client.OperatorsV1alpha1().ClusterServiceVersions(csv.GetNamespace()).Delete(csv.GetName(), &metav1.DeleteOptions{}); err != nil {
-			return err
+	if annotations := parent.GetAnnotations(); annotations != nil {
+		if _, ok := resolver.NewNamespaceSet(strings.Split(annotations[v1alpha2.OperatorGroupTargetsAnnotationKey], ","))[csv.GetNamespace()]; !ok {
```

Comment:
You should probably use the `Contains(...)` method https://github.com/operator-framework/operator-lifecycle-manager/blob/e1e34a0e3acc16d9c48ff31b5f2d44aea0f6dfbb/pkg/controller/registry/resolver/groups.go#L52

It handles all-namespace edge cases too, which might be a problem here.